[DOCS-10758] Consolidate CCM Tags docs (#30578)
DataDog/documentation: content/en/cloud_cost_management/tags/_index.md

| Datadog Enrichment | Cloud Network Monitoring | Source and destination dimensions from the Datadog Cloud Network product. Requires enabling Cloud Network in the Datadog agent |
| Kubernetes Enrichment | Kubernetes Node | User-defined tags found on Kubernetes nodes monitored with Datadog | 
| Kubernetes Enrichment | Kubernetes Pod | User-defined tags found on Kubernetes pods monitored with Datadog | 
| Kubernetes Enrichment | Kubernetes Persistent Volume | User-defined tags found on Persistent Volumes in Kubernetes clusters monitored with Datadog |
| Kubernetes Enrichment | Kubernetes Persistent Volume Claim | User-defined tags found on Persistent Volume Claims in Kubernetes clusters monitored with Datadog | 
| Cloud Cost Management | Cloud Cost Aliases | Some aliases are tags derived from provider cost data, used to simplify the cost data model, such as `aws_product`, which is an alias of `lineItem/ProductCode`. Additional tags that exist on both cost data and integration metrics are added, allowing cost data and usage data to be combined in Custom Allocation Rules, dashboards, and notebooks |

> | Cloud Cost Management | Cloud Cost Allocation | Tags created during cost allocation that specify the split of shared resources, such as `allocated_spend_type` |

Reviewer: maybe a link to [the docs](https://docs.datadoghq.com/cloud_cost_management/cost_allocation/container_cost_allocation/?tab=aws#understanding-spend)

| Cloud Cost Management | FOCUS | Provider-agnostic tags compliant with [FOCUS][8], the unifying format for cloud billing data |
| Tag Pipelines | Rules defined by user | Tags created by applying your Tag Pipelines to cost data |

> | Custom Allocation Rules | Rules defined by user | Tags created by applying your Custom Allocation Rules to cost data (do not apply to SaaS costs) |

Reviewer: links to docs for tag pipelines and allocation rules

Datadog also adds provider specific tags:

| Provider | What tags are collected | Description |
|---|---|---|
| AWS | Cost Allocation Tags | Any tag defined by the user in [AWS Cost Allocation][1] tags that show up in the AWS CUR |

> | AWS | AWS Resource Groups Tagging API | User-defined tags on a cloud resource in AWS, pulled by Cloud Cost Management using the Group Tagging API |

Reviewer: ```suggestion
| AWS | AWS Resource Groups Tagging API | User-defined tags on a cloud resource in AWS, pulled by Cloud Cost Management using the Groups Tagging API |
```

and add this link - https://docs.aws.amazon.com/resourcegroupstagging/latest/APIReference/overview.html

| AWS | AWS Organizations - Accounts | User-defined tags on an AWS account using [AWS Organizations][7] |
| Amazon ECS | Amazon ECS Task | User-defined tags on an ECS task definition |
| Amazon ECS | Amazon ECS Container | User-defined tags on a container running in an ECS task |
| Azure | Azure Cost Export - User Resource Tags | User-defined tags on a cloud resource in Azure, found in the Tags column in the Azure cost export. Does not include Resource Group tags |
| Azure | Azure Cost Export - System Resource Tags | Azure-defined tags on a cloud resource, found in the AdditionalInfo column in the Azure cost export |
| Google Cloud | Google Billing Export - Project Labels| User-defined labels on a project in Google Cloud, found in the project.labels column in the billing export |
| Google Cloud | Google Billing Export - System Resource Labels | System-generated labels on a resource in Google Cloud, found in the system_labels column in the billing export |
| Google Cloud | Google Billing Export - User Resource Labels | User-defined labels on a cloud resource in Google Cloud, found in the labels column in the billing export |

> | Google Cloud | Google Billing Export - User Resource Tags | User-defined tags on a cloud resource in Google Cloud, found in the tags column in the billing export. The `goog-originating-sku-description` tag is added leveraging Google's SKU APis, to provide more granular SKU details for commitment line items. |

Reviewer: `goog-originating-sku-description` is not really a user resource tag... technically isn't that one a billing dimension? but we don't have an obvious spot for it in these tables.

| Google Cloud | GKE Pod | User-defined labels found on pods running in Google Kubernetes Engine | 
| Oracle Cloud | OCI Cost Export - User Resource Tags | User-defined tags on a cloud resource in Oracle Cloud Infrastructure, from the Tags column in the OCI FOCUS cost export |
| Datadog | Datadog Usage Attribution | User-defined tags for Usage Attribution in Datadog Plan and Usage |
| Custom Costs | Cost File Tags | User-defined tags for every provider, found on [cost files][9] uploaded to Cloud Cost Management |

## How tags are normalized

Tag keys and/or values may look slightly different in Cloud Cost Management compared to the providers or other parts of Datadog because of tag normalization. 

Cloud Cost Management normalizes tag **keys** largely the same as Datadog Metrics:
- Drop leading characters that are not letters
- Lower-case all characters
- Replace special characters and spaces with single underscore
- Remove any trailing underscore
- Reduce contiguous underscores to a single underscore
- Tag keys up to 5000 characters are supported, and any characters up until a letter are dropped so that tag keys start with letters (differs from Datadog Metrics)

> Cloud Cost Management normalizes tag **values** as well, while maintaining human readable tag values for cost reporting, with the following logic:

Reviewer: an example of a human readable dimension might be helpful here, as it isn't fully clear what value that brings. like `aws_instance_family:Machine Learning ASIC Instances` is a good example. Not sure how to work it in though...

Author: would this be helpful?

```suggestion
Cloud Cost Management normalizes tag **values** as well, while maintaining human readable tag values for cost reporting. For example, `aws_instance_family:Machine Learning ASIC Instances` remains readable rather than being converted to something like `machine_learning_asic_instances`. The normalization follows this logic:
```

- Convert any consecutive whitespace to a single space
- Keep all letters, marks, numbers, punctuation, and symbols
- Replace any other characters with an underscore `_`
- Tag values up to 5000 characters are supported

## How tags are prioritized

> It's possible for a cost data row to have multiple values, if tag values from 2+ sources are combined, and one is not prioritized over the other.

Reviewer: ```suggestion
It's possible for a cost data row to have multiple values for the same tag key, if tag values from 2+ sources are combined, and one is not prioritized over the other.
```

> To resolve conflicts and mitigate this, Cloud Cost Management will replace existing tags instead of adding duplicates using the most specific source for each tag key. For example, a Kubernetes Pod tag `team:shopist` would take precedence and replace an Kubernetes node tag `team:compute`.

Reviewer: ```suggestion
To resolve conflicts and mitigate this, Cloud Cost Management will replace existing tags instead of adding duplicates using the most specific source for each tag key. For example, a Kubernetes Pod tag `team:shopist` would take precedence and replace a Kubernetes node tag `team:compute`.
```

Sources higher in this list replace tag values from sources lower in this list, if there are conflicts:

> - Tag Pipelines and Custom Allocation Rules

Reviewer: not exactly true, tag pipelines do not replace values, they append.

Custom Allocation Rules do overwrite, i'm pretty sure.

Reviewer 2: this is correct

- FOCUS
- Service Catalog
- Amazon ECS Container
- Amazon ECS Task
- Kubernetes Pod
- Kubernetes Persistent Volume
- Kubernetes Node 
- Host Agent

> All other tag sources (such as bill columns, AWS Organization tags, etc) can be overriden by these sources.

Reviewer: i don't think we let you override bill columns... @elihschiff don't we have some columns that are "reserved" in the merger?

- link: "/cloud_cost_management/tags/tag_pipelines"
  tag: "Documentation"
  text: "Tag Pipelines"
- link: "/cloud_cost_management/tags/tag_explorer"
  tag: "Documentation"
  text: "Tag Explorer"
- link: "/getting_started/tagging/"
  tag: "Documentation"
  text: "Getting Started with Tags"
---

## Overview

Tags help you investigate and understand your cloud and SaaS costs across any dimensions. Tags consist of tag keys and values. For example, in `aws_product:ec2`, the tag key is `aws_product`, and the value is `ec2`.

Cloud Cost Management automatically enriches your cost data with tags from multiple sources, to help you achieve better cost allocation and get deeper insight into who owns infrastructure costs in your ever-changing cloud environments. Using tags, you can allocate shared costs fairly, create accurate reports, and track costs by team, service, or environment. 

## Where tags come from

Across all cloud and SaaS providers, Datadog collects tags from the following sources (including enriching cost data with data from other Datadog products) and adds them to your cost data:

| Source | What tags are collected | Description |
|---|---|---|

> | All Providers | Bill columns | Such as AWS Cost and Usage Report (CUR) columns, Google Billing Export columns, etc |

Author: ```suggestion
| All Providers | Bill columns | Such as AWS Cost and Usage Report (CUR) columns, Google Billing Export columns, and so on |
```

> | All Providers | Bill columns | Such as AWS Cost and Usage Report (CUR) columns, Google Billing Export columns, etc |

Author: oof realized this shouldn't even be etc. just updated.

> | All Providers | Bill columns | Such as AWS Cost and Usage Report (CUR) columns, Google Billing Export columns, etc |

Reviewer 3: Actually, `etc` is OK! It is an exception to our "no latin abbreviations" rule because of how common it is, but this is fine too
